docs/madr/decisions/049-policy-on-namespace.md

```diff
@@ -0,0 +1,681 @@
+
+We don't want to change `spec` on the fly, because it might be surprising for user to see completely different resource in the store.
+Also, in case of the future changes in the logic, it's preferable to store the less explicit version. 
+But we probably have to implement a new feature in the Inspect API that allows viewing the most explicit version of the resource.
+
+#### Empty namespace in `to[].targetRef`
+
+If the `spec.to[idx].targetRef.namespace` is not specified then it's equal to `metadata.namespace`.
+
+#### Empty top-level targetRef
+
+* if the top-level targetRef is not specified in **producer policy**
+  then it's assumed to be `Mesh`
+* if the top-level targetRef is not specified in **consumer policy** or **workload-owner policy**
+  then it's assumed to be `MeshSubset` with `kuma.io/zone` and `k8s.kuma.io/namespace` tags.
+
+The following producer policies semantically equal:
+
+```yaml
+metadata:
+  namespace: backend-ns
+spec:
+  to:
+    - targetRef:
+        kind: MeshService
+        name: backend
+---
+metadata:
+  namespace: backend-ns
+spec:
+  targetRef:
+    kind: Mesh
+  to:
+    - targetRef:
+        kind: MeshService
+        name: backend
+        namespace: backend-ns
+```
+
+The following workload-owner policies are semantically equal:
+
+```yaml
+metadata:
+  namespace: backend-ns
+spec:
+  from:
+    - targetRef:
+        kind: Mesh
+---
+metadata:
+  namespace: backend-ns
+spec:
+  targetRef:
+    kind: MeshSubset
+    tags:
+      k8s.kuma.io/namespace: backend-ns
+      kuma.io/zone: zone-with-backend
+  from:
+    - targetRef:
+        kind: Mesh
+```
+
+The following consumer policies are semantically equal:
+
+```yaml
+metadata:
+  namespace: frontend-ns
+spec:
+  to:
+    - targetRef:
+        kind: MeshService
+        name: backend
+        namespace: backend-ns
+---
+metadata:
+  namespace: frontend-ns
+spec:
+  targetRef:
+    kind: MeshSubset
+    tags:
+      k8s.kuma.io/namespace: frontend-ns
+      kuma.io/zone: zone-with-frontend
+  to:
+    - targetRef:
+        kind: MeshService
+        name: backend
+        namespace: backend-ns
+```
+
+#### Empty `from[].targetRef`
+
+Most of the policies support only `kind: Mesh` in `from[].targetRef` so it makes sense to allow the field to be empty.
+If `from[].targetRef` is not specified we're going to assume it's equal to `kind: Mesh`.
+The following policies are semantically equal:
+
+```yaml
+metadata:
+  namespace: backend-ns
+spec:
+  from:
+    - default: {}
+---
+metadata:
+  namespace: backend-ns
+spec:
+  targetRef:
+    kind: MeshSubset
+    tags:
+      k8s.kuma.io/namespace: backend-ns
+      kuma.io/zone: zone-with-backend
+  from:
+    - targetRef:
+        kind: Mesh
+      default: {}
+```
+
+### Namespace-scoped policies and multizone
+
+Namespace-scoped policies can be applied to a custom namespace only on zones as there are simply no DPPs and custom
```

Comment:
Because we're syncing policies to other zones now, maybe there should be something in here about where/if egress/ingress play a role.

Reply:
Ingress is never modified by the policies, so I guess it doesn't play role in syncing. Egress can be affected by ExternalServices, so we probably have to add something about the ExternalServices here

Reply:
MeshExternalService can be created only in `kuma-system` namespace. This means, all policies that reference MeshExternalService are consumer policies and shouldn't be synced. I'm going to put this into the doc

Reply:
I added:

> Policies that reference `MeshExternalService` in `to[].targetRef` won't be synced to other zones. MeshExternalService can be created only in `kuma-system` namespace and that's why the policy it's referenced by is always a consumer policy. That's why the policies that have potential to indirectly change ZoneEgress configuration are never synced to other zones.